ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/catalog.py

```diff
@@ -0,0 +1,132 @@
+
+        Raises:
+            AristaAvdError: Will raise an error if the input catalog cannot be loaded or is empty.
+
+        """
+        if custom_catalog is not None:
+            if not isinstance(custom_catalog, dict):
+                raise AristaAvdError("The provided custom_catalog is not a dictionary.")
+            return custom_catalog
+
+        elif custom_catalog_path is not None:
+            try:
+                with open(custom_catalog_path, "r", encoding="UTF-8") as file:
+                    input_catalog = safe_load(file) or {}
+            except Exception as error:
+                raise AristaAvdError(f"Exception raised while processing file {custom_catalog_path}. Details: {str(error)}") from error
+
+            return input_catalog
+
+        return {}
+
+    def _default_catalog(self) -> dict:
+        """
+        Create the default catalog based on the AVD test classes.
+
+        Test definitions are generated from the AVD structured_config for each AVD test classes and are merged together to create the final default catalog.
+
+        Tests can be skipped from the default catalog depending on the self.skipped_tests attribute.
+
+        Returns:
+            dict: The default catalog generated from AVD_TEST_CLASSES.
+        """
+        default_catalog = {}
+
+        # TODO maybe it should require device_name
+        eos_validate_state_vars = get_eos_validate_state_vars(self.hostvars)
+
+        for avd_test_class in AVD_TEST_CLASSES:
+            # Check if the whole class is to be skipped
+            # TODO - need to explain the syntax of this dictionary with care
+            if avd_test_class.__name__ in self.skipped_tests and self.skipped_tests[avd_test_class.__name__] is None:
+                continue
+
+            # Initialize the test class
+            eos_validate_state_module: AvdTestBase = avd_test_class(hostvars=self.hostvars, custom_data=eos_validate_state_vars, device_name=self.device_name)
+            generated_tests = eos_validate_state_module.render()
+
+            # Remove the individual tests that are to be skipped
+            # TODO - this should be possuble to simplify this
+            if (avd_test_class_skipped_tests := self.skipped_tests.get(avd_test_class.__name__)) is not None:
+                for anta_tests in generated_tests.values():
+                    anta_tests[:] = [test for test in anta_tests if list(test.keys())[0] not in avd_test_class_skipped_tests]
+
+            default_catalog = self.merge_catalogs(default_catalog, generated_tests)
+
+        return default_catalog
+
+    @staticmethod
+    def merge_catalogs(*args: dict) -> dict:
+        """
+        Utility function to merge catalogs according to the ANTA catalog structure.
+
+        Args:
+            *args (dict): Catalogs to merge.
+
+        Returns:
+            dict: The merged catalog.
+        """
+        merged_catalog = {}
+        for catalog in args:
```

Comment:
I think `deepmerge.always_merger(merged_catalog, *args, recursive=True)` could do this.

Reply:
Hmmm - that is probably the case that deepmerge has something for us but there is no `recursive` keyword in always_merger. Will look into it